feat: create mint function/interface

imqdee opened the pull request.

imqdee committed:
feat: develop mint logic

imqdee committed:
chore: install/setup `@ledgerhq/react-ui` package

- Create custom `_document` to manage styled-component server side rendering
- Override default Nextjs babel config to manage styled-components
- Add the UI provider to the `_app` component and set the theme as dark
- Create global styles to set the background color of the dark mode
- Add all the fonts needed by the package
- Add these library
--- @ledgerhq/react-ui
--- styled-components
--- babel-plugin-styled-components

imqdee committed:
chore: moving pages/ to src/

imqdee committed:
chore: setup custom page extension

That will allow me to create non-page file insides pages directory. I'll
use this possibility to create page sections for example.

CremaFR reviewed (approved):
LGTM, only small comment that can be managed later

imqdee committed:
design: create generic header

- Create connector component to connect/disconnect to the provider
- Create generic header and inject it into the _app component
- Add 2 logos (Metamask and walletConnect)
- Switch back to the light mode and remove generic stylesheet

imqdee committed:
feat(admin): develop the recipient section

- Integrate basic style for the admin page
- Develop the entiere recipient section
- Update typescript configuration to allow down level iteration. That
allow you to iterate on set() or string type
- Create a getRandomColorPerAddress function that takes an address as
parameter and return an unique color for each address

imqdee committed:
feat(admin): develop the token section

- Develop the entiere token section
- Create a json file called stablecoins that stores some stablecoins addresses
for specific networks
- Move contracts.json file at the freshly created folder called data

imqdee committed:
feat(admin): develop the amount section

- Develop the entiere amount section
- Improve generic layout in _app
- Fix disabled propriety for the mint button in the admin page

imqdee committed:
feat(admin): develop the confirmation section

- Clean the admin page
- Create the confirmation section
- Create an utils function to preview arbitrary hash

imqdee committed:
refactor: create custom hook to mint

This custom hook, called `useVoucherMint` has the responsibility to call
the `mint` or the `batchMint` method regarding the number of recipients
passed.
